src/main.rs

```diff
@@ -400,26 +400,12 @@ impl SpiDrv {
         }
     }
 
-    fn send_param_string(&mut self, uart: &mut EnabledUart, param_string: String<STR_LEN>, last_param: bool) -> SpiResult<()> {
-      let mut i: u8 = 0;
-      let mut param_bytes: String<STR_LEN> = param_string.into_bytes().iter().rev().collect();
-      loop {
-        let byte = match param_bytes.pop() {
-            Some(byte) => byte,
-            None => return Ok(())
-        };
-        let mut param: Params = Params::new();
-        param.push(byte).unwrap();
-        let result = self.send_param(uart, param, 1, last_param);
-        // Exit when we reach a '\n':
-        if byte == 0xD {
-           return result;
-        }
-        i += 1;
-        if i == STR_LEN as u8 {
-           return result;
-        }
-      }
+    fn send_param_string(&mut self, uart: &mut EnabledUart, param_string: String<STR_LEN>, last_param: bool) -> () {
+        param_string.into_bytes().iter().for_each(|byte| {
+            let mut param: Params = Params::new();
+            param.push(*byte).unwrap();
+            self.send_param(uart, param, 1, last_param);
```

Comment:
@calebbourg This looks amazing, very nice work simplifying and making a better function than the original version!

A couple of issues to point out that make this function not yet function the way we need it to:

1. It doesn't stop sending bytes when it detects the '\n' character
2. If the call to `send_param_string()` does set `last_param` to `true` it will send out an `END_CMD` byte for every single character in the string - instead, we only want that to be sent out once for the very last string character

Reply:
Nicely refactored, resolving these comments.